src/tools/wallmap/objloader.go

```diff
@@ -0,0 +1,194 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"math"
+	"os"
+	"strings"
+)
+
+type Point2 struct {
+	X, Y float64
+}
+
+func (p *Point2) Scale(f float64) {
+	p.X = p.X * f
+	p.Y = p.Y * f
+}
+
+func Point2From3(p Point3) Point2 {
+	return Point2{p.X, p.Y}
+}
+
+type Point3 struct {
+	X, Y, Z float64
+}
+
+func (p *Point3) Set(s []string) error {
+	if _, err := fmt.Sscanf(s[0], "%f", &p.X); err != nil {
+		return fmt.Errorf("invalid syntax \"%s\"", s)
+	}
+	if _, err := fmt.Sscanf(s[1], "%f", &p.Z); err != nil {
+		return fmt.Errorf("invalid syntax \"%s\"", s)
+	}
+	if _, err := fmt.Sscanf(s[2], "%f", &p.Y); err != nil {
+		return fmt.Errorf("invalid syntax \"%s\"", s)
+	}
+	return nil
+}
+
+type Triangle struct {
+	P1, P2, P3 Point2
+}
+
+func (t *Triangle) Scale(f float64) {
+	t.P1.Scale(f)
+	t.P2.Scale(f)
+	t.P3.Scale(f)
+}
+
+func (t Triangle) MinX() float64 {
+	return math.Min(t.P1.X, math.Min(t.P2.X, t.P3.X))
+}
+
+func (t Triangle) MinY() float64 {
+	return math.Min(t.P1.Y, math.Min(t.P2.Y, t.P3.Y))
+}
+
+func (t Triangle) MaxX() float64 {
+	return math.Max(t.P1.X, math.Max(t.P2.X, t.P3.X))
+}
+
+func (t Triangle) MaxY() float64 {
+	return math.Max(t.P1.Y, math.Max(t.P2.Y, t.P3.Y))
+}
+
+func (t Triangle) isDegenerate() bool {
+	area := ((t.P2.X-t.P1.X)*(t.P3.Y-t.P1.Y) -
+		(t.P3.X-t.P1.X)*(t.P2.Y-t.P1.Y))
+	// TODO: also check area with an epsilon?
+	return area == 0.0
+}
+
+type ObjFile struct {
+	Vertices               []Point2
+	Triangles              []Triangle
+	MinX, MinY, MaxX, MaxY float64
+	dbg                    bool
+}
+
+// SetMin checks if a > b, then a will be set to the value of b.
+func SetMin(a *float64, b float64) {
+	if b < *a {
+		*a = b
+	}
+}
+
+// SetMax checks if a < b, then a will be set to the value of b.
+func SetMax(a *float64, b float64) {
+	if *a < b {
+		*a = b
+	}
+}
+
+func (of *ObjFile) parseVertex(lineno int, kw string, data []string) error {
+	p3 := Point3{}
+	err := p3.Set(data)
+	if err != nil {
+		return err
+	}
+	// discard the Z coordinate
+	of.Vertices = append(of.Vertices, Point2From3(p3))
+	return nil
+}
+
+func (of *ObjFile) parseFace(lineno int, kw string, data []string) error {
```

Comment:
As far as I can tell `lineno` is only useful for error reporting. I'd skip it: let this function returns whatever errors and you can wrap them in the callers like: `fmt.Errorf("line %d: error parsing face: %v", lineno, err)`